### PR TITLE
[MA-681] Implement Universal Debug

### PR DESCRIPTION
**Pull request by mikeshutov-bolt**
# Description
Implement Universal Debug to return only requested debug information or specific log file.

Fixes: [MA-681](https://boltpay.atlassian.net/browse/MA-681)

#changelog [WIP] Implement Universal Debug

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Comment by njfaries** [comment]
@mikeshutov-bolt is this still WIP?

**Comment by mikeshutov-bolt** [comment]
> @mikeshutov-bolt is this still WIP?

Yes need to work on tests.

File: Model/Api/Debug.php [hunk 1]
```diff
@@ -198,11 +197,18 @@ public function universalDebug($type)
...
-        throw new BoltException(
-            __('Webhook of type Debug has not been implemented'),
-            null,
-            BoltErrorResponse::ERR_SERVICE
-        );
+        $result = $this->debugInfoFactory->create();
+        switch ($data['type']){
+            case 'log':
+                $result->setLogs($this->logRetriever->getLogOfType($data['log_type']));
+                break;
```
Comment (njfaries) [inline Model/Api/Debug.php:204]: Do we need to test these keys for existing?

**Review by njfaries — APPROVED** [review]

File: Model/Api/Debug.php [hunk 2]
```diff
@@ -198,11 +197,18 @@ public function universalDebug($type)
...
-        // Throw debg not implemented exception for now untill it is implemented.
-        throw new BoltException(
-            __('Webhook of type Debug has not been implemented'),
-            null,
-            BoltErrorResponse::ERR_SERVICE
-        );
+        $result = $this->debugInfoFactory->create();
+        switch (isset($data['type']) ? $data['type'] : ''){
+            case 'log':
+                $result->setLogs($this->logRetriever->getLogOfType(isset($data['log_type']) ? $data['log_type'] : null));
```
Comment (njfaries) [inline Model/Api/Debug.php:203]: What happens if null is passed to `getLogOfType`? Does it throw an exception of some kind or just return an empty result?

Comment (mikeshutov-bolt) [reply Model/Api/Debug.php:203]: Returns default logs.

**Pull request merged**